packages/react/src/components/TextField/TextField.tsx

```diff
@@ -16,29 +16,108 @@
  * under the License.
  */
 
-import {InputLabel} from '@mui/material';
+import InputAdornment from '@mui/material/InputAdornment';
 import MuiTextField, {TextFieldProps as MuiTextFieldProps} from '@mui/material/TextField';
+import {DoubleCircleIcon, VisibilityIcon, VisibilityOffIcon} from '@oxygen-ui/react-icons';
 import clsx from 'clsx';
-import {FC, ReactElement} from 'react';
+import {FC, MouseEvent, ReactElement, ReactNode, useState} from 'react';
 import {WithWrapperProps} from '../../models';
 import {composeComponentDisplayName} from '../../utils';
+import IconButton from '../IconButton';
+import InputLabel from '../InputLabel';
+import List from '../List';
+import ListItem from '../ListItem';
+import ListItemIcon from '../ListItemIcon';
+import ListItemText from '../ListItemText';
+import Tooltip from '../Tooltip';
 import './text-field.scss';
 
-export type TextFieldProps = MuiTextFieldProps;
+export type TextFieldProps = {
+  /**
+   * Criteria for user input.
+   */
+  criteria?: string[];
+} & MuiTextFieldProps;
 
 const COMPONENT_NAME: string = 'TextField';
 
 const TextField: FC<TextFieldProps> & WithWrapperProps = (props: TextFieldProps): ReactElement => {
-  const {className, label, InputLabelProps, id, ...rest} = props;
+  const {className, criteria, InputLabelProps, id, label, type, ...rest} = props;
 
   const classes: string = clsx('oxygen-text-field', className);
 
+  const [open, setOpen] = useState<boolean>(false);
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handleClick = (): void => {
+    if (type === 'password' && criteria?.length > 0) {
```

Comment:
isn't there an enum or a constant for the type?

Reply:
Check the `HTMLInputElement` interface.

Reply:
MUI doesn't have any exported enums or constants for input types. Since the types are the same as the standard HTML5 input types  will they tend to change? In MUI docs they have used the string literals. wdyt?

Reply:
Added a constant file for `TextField` folder